drape_frontend/stylist.cpp

```diff
@@ -144,21 +154,34 @@ class Aggregator
   void Init()
   {
     m_featureLayer = m_f.GetLayer();
-    // @todo m_priorityModifier is not used, try to apply to areas to render in size order.
-    if (m_geomType == feature::GeomType::Point)
-      m_priorityModifier = (double)m_f.GetPopulation() / 7E9;
-    else
+    if (m_geomType == feature::GeomType::Area)
     {
+      // Calculate depth based on areas' sizes instead of style-set priorities.
       m2::RectD const r = m_f.GetLimitRect(m_zoomLevel);
-      m_priorityModifier = std::min(1.0, r.SizeX() * r.SizeY() * 10000.0);
+
+      // Raw areas' size range of about (1e-10, 3000) is too big, have to shrink it.
+      double const areaSize = r.SizeX() * r.SizeY();
+      // log2() of numbers <1.0 is negative, adjust it to be positive.
+      double const minLog2 = std::log2(std::numeric_limits<float>::min()); // == -126
+      double const areaSizeCompact = std::log2(areaSize) - minLog2;
+      // Should be well below lines and foreground areas for the layering logic to work correctly,
+      // produces a depth range of about (-13000, -8000).
+      m_areaDepth = kMinLinesDepth - areaSizeCompact * 100.0f;
+
+      // +/-20000 seems to be depth limits imposed by OpenGL (?) out of which areas won't render.
+      double const minDepth = -20000.0f;
+      ASSERT_GREATER_OR_EQUAL(m_areaDepth, minDepth, (m_areaDepth));
```

Comment:
We already have needed constants in drape/utils/projection.hpp
ASSERT(dp::kMinDepth < m_areaDepth && m_areaDepth < dp::kMaxDepth, (m_areaDepth));
with comments

Reply:
Done